code/game/machinery/_machinery.dm

```diff
@@ -261,6 +261,22 @@ Class Procs:
 			L.update_mobility()
 	set_occupant(null)
 
+/**
+ * Puts passed object in to user's hand
+ *
+ * Puts the passed object in to the users hand if they are adjacent.
+ * If the user is not adjacent then place the object on top of the machine.
+ *
+ * Vars:
+ * * object (obj) The object to be moved in to the users hand.
+ * * user (mob/living) The user to recive the object
+ */
+/obj/machinery/proc/try_put_in_hand(obj/object, mob/living/user)
+	if(!issilicon(user) && in_range(src, user))
+		user.put_in_hands(object)
+	else
+		object.forceMove(drop_location())
```

Comment:
Ideally this shouldn't check for silicons but instead use the failure state of put in hands to determine if it failed or not. It should also use CanReach so that you can't use it through one-way glass panes.
```
if (!user.CanReach(src) || !user.put_int_hands(object))
  object.forceMove(drop_location())
```

Reply:
tweaked in 673a2d1